js/src/BuilderSettingsMenu.js

```diff
@@ -0,0 +1,409 @@
+/**
+ * SettingsMenu
+ */
+
+/** @jsx h */
+import { h, Component } from 'preact'
+import '../../css/src/SettingsMenu.css'
+
+const ScaleEditor = require('./ScaleEditor')
+
+class BuilderSettingsMenu extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      display: props.display
+    }
+    props.settings.hold_changes()
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.setState({display: nextProps.display})
+    nextProps.settings.hold_changes()
+  }
+
+  abandonChanges () {
+    this.props.settings.abandon_changes()
+    this.setState({display: 'none'})
+  }
+
+  saveChanges () {
+    this.props.settings.accept_changes()
+    this.setState({display: 'none'})
+  }
+
+  handleStyle (value, type) {
+    let style = []
```

Comment:
If `style` won't be overwritten turn it into a `const`

Reply:
Done